src/mon/Monitor.cc

```diff
@@ -3697,6 +3700,59 @@ void Monitor::handle_command(MonOpRequestRef op)
     rdata.append(ds);
     rs = "";
     r = 0;
+  } else if (prefix == "mon ok-to-stop") {
+    vector<string> ids;
+    if (!cmd_getval(g_ceph_context, cmdmap, "ids", ids)) {
+      r = -EINVAL;
+      goto out;
+    }
+    set<string> wouldbe;
+    for (auto rank : quorum) {
+      wouldbe.insert(monmap->get_name(rank));
+    }
+    for (auto& n : ids) {
+      if (monmap->contains(n)) {
+	wouldbe.erase(n);
+      }
+    }
+    ostringstream ss;
+    if (wouldbe.size() < monmap->size() / 2 + 1) {
```

Comment:
the common sense about quorum is repeating itself in different places. could we have some helpers in `MonMap` to define it? like
```
unsigned min_quorum_size() const {
  return min_quorum_size_of(size());
}
static unsigned min_quorum_size_of(unsigned size) {
  return size / 2 + 1;
}
```
and update `Monitor::handle_probe_reply()` accordingly.

Reply:
added